org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/SourceAssistProcessor.java

```diff
@@ -135,21 +137,20 @@ public SourceAssistProcessor(PreferenceManager preferenceManager) {
 			addSourceActionCommand($, params.getContext(), overrideMethods);
 		}
 
-		// Generate Getter and Setter
-		Optional<Either<Command, CodeAction>> getterSetter = getGetterSetterAction(params, context, type);
-		addSourceActionCommand($, params.getContext(), getterSetter);
+		// Generate Getter and Setter QuickAssist
+		if (isInTypeDeclaration(context)) {

```

Comment:
isInTypeDeclaration(context) is called multiple times in source action handler, you can assign it to a local variable `boolean isInTypeDeclaration = isInTypeDeclaration(context);` for reuse.

Reply:
Sure.